pkg/feature/manifest.go

```diff
@@ -1,76 +1,103 @@
 package feature
 
 import (
+	"bytes"
+	"embed"
 	"fmt"
 	"html/template"
-	"os"
+	"io"
+	"io/fs"
 	"path/filepath"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
-const BaseOutputDir = "/tmp/odh-operator"
+//go:embed templates
+var embeddedFiles embed.FS
 
 type manifest struct {
 	name,
-	path string
+	path,
+	processedContent string
 	template,
-	patch,
-	processed bool
+	patch bool
 }
 
-func loadManifestsFrom(path string) ([]manifest, error) {
+func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
 	var manifests []manifest
-	if err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+
+	err := fs.WalkDir(fsys, path, func(path string, dirEntry fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
-		if info.IsDir() {
+
+		if dirEntry.IsDir() {
 			return nil
 		}
-		basePath := filepath.Base(path)
-		manifests = append(manifests, manifest{
-			name:     basePath,
-			path:     path,
-			patch:    strings.Contains(basePath, ".patch"),
-			template: filepath.Ext(path) == ".tmpl",
-		})
+
+		_, err = fs.ReadFile(fsys, path)
+		if err != nil {
+			return err
+		}
```

Comment:
I do not think we need to read the file here, as we anyway do not use the content. To optimize it, I was thinking we can do the following instead:

```golang
_, err = dirEntry.Info()
if err != nil {
	return err
}

if dirEntry.IsDir() {
	return nil
}
```

(I can't really `suggest` this as it changes the order of already existing code)

Calling `Info` will result in calling `fstat` underneath. This is enough to capture errors such as "file not found", "permission denied", "general i/o" in advance.

Reply:
good catch, done here [54f4f38](https://github.com/opendatahub-io/opendatahub-operator/pull/797/commits/54f4f3835be46134b21a2ec5dba81e2517bfe93d)